### DISCUSSION TITLE
[Question]: How to change endpoint from v1 to v1beta for gemini?

### DISCUSSION THREAD

**Discussion by TheValkyrja** [danny-avila/LibreChat#2378] [Q&A]
### What is your question?

is there any way to set the endpoint for gemini from v1 to v1beta?

### More Details

i just gained the access to gemini 1.5 api, which is availible through generativelanguage.googleapis.com/v1beta/models/.
i'd like to access the model with LibreChat but it seems only sending request to /v1.
is there ways for me to configure the endpoint? (in googlegenerative etc.)

### What is the main subject of your question?

Endpoints

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by mcwongmc** [comment]
Same here. I think a lot of people will raise the issue as Gemini 1.5 pro api is out. And it has free tier also. Thanks.

https://developers.googleblog.com/2024/04/gemini-15-pro-in-public-preview-with-new-features.html

**Comment by danny-avila** [comment]
It will be supported soon after public release

**Comment by ath** [comment]
Is there now support for `gemini-2.0-pro-exp-02-05` and `gemini-2.0-flash-thinking-exp-01-21`?

2025-03-05 00:50:52 error: [handleAbortError] AI response error; aborting request: [GoogleGenerativeAI Error]: Error fetching from https://generativelanguage.googleapis.com/v1/models/gemini-2.0-pro-exp-02-05:streamGenerateContent?alt=sse: [404 Not Found] models/gemini-2.0-pro-exp-02-05 is not found for API version v1, or is not supported for generateContent. Call ListModels to see the list of available models and their supported methods.

**Comment by ath** [comment]
Okay, the solution is described here:
https://github.com/danny-avila/LibreChat/discussions/6072

I deleted some older Docker images, pulled from git, pulled the updated/new Docker images, restarted, and now Gemini is working correctly.
